__tests__/index.js

```diff
@@ -29,31 +28,33 @@ describe('probot-app-merge-pr', () => {
     app.app = () => 'test';
   });
 
+  function makeMergeRequest() {
+    return new Promise(resolve => {
+      nock('https://api.github.com')
+        .get('/repos/fusionjs/test-repo/collaborators/test-user/permission')
+        .reply(200, fixtures.reviewUserPermissionLevel)
+        .get('/repos/fusionjs/test-repo/contents/.github/merge-pr.yml')
+        .reply(404)
+        .get('/repos/fusionjs/test-repo/pulls/1/commits')
+        .reply(200, fixtures.listCommits)
+        .put('/repos/fusionjs/test-repo/pulls/1/merge', resolve)
+        .reply(200);
+    });
+  }
+
   test('merges the PR with commit authors as co-authors', async () => {
-    expect.assertions(1);
-
-    nock('https://api.github.com')
-      .get('/repos/fusionjs/test-repo/collaborators/test-user/permission')
-      .reply(200, fixtures.reviewUserPermissionLevel)
-      .get('/repos/fusionjs/test-repo/contents/.github/merge-pr.yml')
-      .reply(404)
-      .get('/repos/fusionjs/test-repo/pulls/1/commits')
-      .reply(200, fixtures.listCommits)
-      .put('/repos/fusionjs/test-repo/pulls/1/merge', body => {
-        const expectedMessage = `https://github.com/fusionjs/test-repo/pull/1
+    const expectedMessage = `https://github.com/fusionjs/test-repo/pull/1
 
 Co-authored-by: test-user2 <test-user2@uber.com>
 Co-authored-by: test-user3 <test-user3@uber.com>`;
 
-        expect(body).toEqual({
-          commit_title: 'Test PR',
-          commit_message: expectedMessage,
-          merge_method: 'squash',
-        });
-        return true;
-      })
-      .reply(200);
+    const awaitMergeRequestPromise = makeMergeRequest();
```

Comment:
Not sure if it's necessary to break this out into a seperate function. Also, that chain of requests is specific to this test. Maybe instead do something like this?

```js
nock('https://api.github.com')
  .get('/repos/fusionjs/test-repo/collaborators/test-user/permission')
  .reply(200, fixtures.reviewUserPermissionLevel)
  .get('/repos/fusionjs/test-repo/contents/.github/merge-pr.yml')
  .reply(404)
  .get('/repos/fusionjs/test-repo/pulls/1/commits')
  .reply(200, fixtures.listCommits);

const mergeRequest = new Promise(resolve => {
  nock('https://api.github.com')
    .put('/repos/fusionjs/test-repo/pulls/1/merge', resolve)
    .reply(200);
});

//...
expect(await mergeRequest)...
```

Reply:
Gotcha, I went ahead and inlined it for now, and we can abstract part of the chain if necessary, but it is less code, and a smaller change.